src/libutil/or-suggestions.hh

```diff
@@ -0,0 +1,64 @@
+#include "suggestions.hh"
+#include "error.hh"
+
+namespace nix {
+
+// Either a value of type `T`, or some suggestions
+template<typename T>
+class OrSuggestions {
```

Comment:
I'm not sure if this class adds much value over plain `std::variant<T, Suggestions>`. In fact it seems to add more cognitive overhead over the `std::variant`...

Reply:
I started with a `std::variant`, but it was a bit annoying because it’s not possible to add methods to it, meaning that I had to resort to inline `std::visit(overload{})`or `std::get_if` everywhere, which was quite unfriendly.

Alternatively, I can switch back to a plain variant, and add some plain functions working on it (which is visually a bit less nice, but avoids this wrapping)

Reply:
Just change it into an assertion since it's clearly a logic error if that happens.

Reply:
Right, makes sense.

I’ve actually replaced the whole thing by a call to `std::get` which will raise its own exception if the bad variant is provided